.github/workflows/integration-redis.yml

```diff
@@ -0,0 +1,15 @@
+name: Redis Integration Tests
```

Comment:
@cijothomas Check it out... a GitHub Action for spinning up a container with Redis and running the integration tests! I'll do one for the .NET Core SqlClient instrumentation after this PR.

Reply:
We have unit tests as part of the other workflows, so do we need to make this as a mandatory step for merge? Or this can be optional? If this is going finish around same time as other, then we can make this a required test. Else we can make it optional, and have folks with "merge" power use their discretion and decide if want to wait for this before merging. (typically for minor PRs like readme. etc)

Reply:
It looks fast enough for Redis. (probably because Redis image is few 100mb) For SQL, this could be GBs and could be slow.

Reply:
To be honest, I have no idea. Not sure if GitHub will download the image every time it is executed, or reuse it somehow. This is all new territory for me! The current incarnation is fast. I think when I ran it, the docker part finished before the other stuff. Preferrable to have integration tests running every time, right? But we could also have a release pipeline that runs integration tests before a formal build/release to NuGet. That's really the important part.

Reply:
The GH Actions runs on "vms" that are thrown away after each job I believe. So no caching.
I noticed Redis part was very fast - My guess is its due to the small size of redis image. SqlServer may not be that. 
Anyway, I am happy to make this required for merge, after letting this run for 3-4 days.